experimental/trusted_shuffler/trusted_shuffler/src/tests.rs

```diff
@@ -27,40 +27,52 @@ impl RequestHandler for TestRequestHandler {
     // A test function that processes requests.
     // In the real use-case, this function should send requests to the backend and
     // return responses.
-    async fn handle(&self, request: Vec<u8>) -> anyhow::Result<Vec<u8>> {
-        if drop_request(request.clone()) {
+    async fn handle(
+        &self,
+        request: TrustedShufflerRequest,
+    ) -> anyhow::Result<TrustedShufflerResponse> {
+        if drop_request(&request) {
             // In that case we don't send an answer ever.
             loop {
                 tokio::task::yield_now().await;
             }
         }
-        Ok(generate_response(request))
+        Ok(generate_response(&request))
     }
 }
 
 // Non-async version of the `handle_request` function used to create expected responses.
-fn generate_response(request: Vec<u8>) -> Vec<u8> {
-    let parsed_request = String::from_utf8(request).expect("Couldn't parse request");
-    format!("Response for: {}", parsed_request).into_bytes()
+fn generate_response(request: &TrustedShufflerRequest) -> TrustedShufflerResponse {
+    TrustedShufflerResponse {
+        body: request.body.clone(),
+    }
 }
 
 // Generates request which will be dropped by the test backend, because we indicate it has to be
 // dropped in the request itself.
-fn generate_dropped_request() -> Vec<u8> {
+fn generate_dropped_request() -> TrustedShufflerRequest {
```

Comment:
Why do we need these drop requests? Is it for testing timeouts?

Reply:
Yes!